test/common/markdown/markdown.tests.ps1

```diff
@@ -7,101 +7,112 @@ Write-Verbose -message "PSScriptRoot: $PSScriptRoot" -Verbose
 $repoRootPathFound = $false
 
 Describe 'Common Tests - Validate Markdown Files' -Tag 'CI' {
-    if (Get-Command -Name 'npm' -ErrorAction SilentlyContinue)
-    {
-        Write-Warning -Message "NPM is checking Gulp is installed. This may take a few moments."
-
-        $null = Start-Process `
-            -FilePath "npm" `
-            -ArgumentList @('install','--silent') `
-            -Wait `
-            -WorkingDirectory $PSScriptRoot `
-            -PassThru `
-            -NoNewWindow
-        $null = Start-Process `
-            -FilePath "npm" `
-            -ArgumentList @('install','-g','gulp','--silent') `
-            -Wait `
-            -WorkingDirectory $PSScriptRoot `
-            -PassThru `
-            -NoNewWindow
-
-        It "Should not have errors in any markdown files" {
-
-            $mdErrors = 0
-            Push-Location -Path $PSScriptRoot
-            try
-            {
-                $docsToTest = @(
-                    './*.md'
-                    './docs/installation/*.md'
-                )
-                $filter = ($docsToTest -join ',')
-                Write-Verbose "Filter: $filter" -Verbose
-                &"gulp" test-mdsyntax --silent `
-                    --rootpath $repoRootPath `
-                    --filter $filter
-
-                Start-Sleep -Seconds 3
-            }
-            catch [System.Exception]
+    BeforeAll {
+        $skip = !$IsWindows
+        if ( !$skip ) 
+        {
+            $NpmInstalled = "not installed"
+            if (Get-Command -Name 'npm' -ErrorAction SilentlyContinue) 
             {
-                Write-Warning -Message ("Unable to run gulp to test markdown files. Please " + `
-                                        "be sure that you have installed nodejs and have " + `
-                                        "run 'npm install -g gulp' in order to have this " + `
-                                        "text execute.")
-            }
-            finally
+                $NpmInstalled = "Installed"
+                Write-Verbose -Message "NPM is checking Gulp is installed. This may take a few moments." -Verbose
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','-g','gulp','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+            } 
+            elseif( -not $env:AppVeyor)
             {
-                Pop-Location
+                Write-Warning "Node and npm are required to run this test"
+                $skip = $true
```

Comment:
We should not skip because we skip  `It` block and `$NpmInstalled | should BeExactly "Installed"` too.
If we don't initialize `$NpmInstalled` here then below `$NpmInstalled | should BeExactly "Installed"` return uninformative message. So we should replace `Write-Warning ` with 
```powershell
$NpmInstalled = "Node and npm are required to run this test" 
```